Usability fixes for ltiprovider

eqsoft opened the pull request:
Hi, these bugfixes and usability fixes needs further discussion because they affect also the new MemberView (Fabian) and LMPresentationGUI (Alex).

Amstutz commented:
Nice:
![image](https://user-images.githubusercontent.com/1866896/70420985-2a318c80-1a69-11ea-9ef1-a5ef056e5464.png)

@eqsoft : If possible, you could remove out commented lines of code, which would make my the above statistic even better ;-)

chfsx commented on Services/Container/classes/Screen/MemberViewLayoutProvider.php:
```diff
@@ -36,7 +36,7 @@ public function isInterestedInContexts() : ContextCollection
     public function getPageBuilderDecorator(CalledContexts $screen_context_stack) : ?PageBuilderModification
     {
-        if (!$screen_context_stack->current()->hasReferenceId()) {
+        if (!$screen_context_stack->current()->hasReferenceId() || $this->dic["lti"]->isActive()) {
```
I'm not a fan of being dependent on LTI-Code here. Isn't there another solution? For example, if we extend the GlobalScreen so that you can not only log in for contexts, but also explicitly log out?

eqsoft replied:
I see that issue but i am not sure if i understand what you mean with a "GS logout" extension . The intention of transfering the responsibility of memberview "rendering" to the LTI PageBuilderModifiction was to avoid a priority collision with LTI Implementation. The base intention of using a PageBuilderModifiction instead of single MetaBar/MainBar/Footer in LTI was to avoid the collision with MainBarModification with the new Help Service.

By taking a loook to the new ModeInfo in MemberView i wonder why this is rendered by the whole page component (page->withModeInfo()) even though Mainbar, Metabar and footer are not affected. Maybe it would be sufficient to surround the content frame content->withModeInfo() for the memberview?, or why not an implementation of flexible ModeInfo: content->withModeInfo page->withModeInfo.. just my thoughts.

chfsx commented on Services/LTI/classes/Screen/LtiViewLayoutProvider.php:
```diff
@@ -18,75 +27,116 @@
-        }
-        else {
-            return null;
+            $this->globalScreen()->layout()->meta()->addCss('./Services/LTI/templates/default/lti.css');
```
this is not necessarily my business, but why does LTI need its own CSS?

Amstutz replied:
I saw this as well, as far as I undwerstood, this seems to be leftover, of some old LTI CSS, right? If not, this seems like an issue.

eqsoft replied:
The (small) injected css is just a tiny hack to discard a (null) footer completely. Without an additional display:none the footer is empty but still displayed. In a Kioskmode the footer and btw. other null modifications (Metabar, Mainbar) should hide the components. What do you mean?

chfsx commented on Services/LTI/classes/Screen/LtiViewLayoutProvider.php:
```diff
@@ -18,75 +27,116 @@
+                    array('mv' => 0)
+                ));
+            }
+            return $this->factory->page()->withHighPriority()->withModification(
```
do I understand correctly that the problem here is that you want to replace the page completely, but an activated MemberView gets lost?

eqsoft replied:
No. with this implementation a course-admin will be able to activate and deactivate the memberview (ModeInfo) within any LTI sessions in the same way as a normal ILIAS course-admin session. That is exactly what we want in LTI :-)

chfsx commented on Services/LTI/classes/Screen/LtiViewLayoutProvider.php:
```diff
@@ -18,75 +27,116 @@
+            ->withSymbol($icon)
+            ->withTitle($title)
+            ->withAction($link);
+        $renderer = new TopLinkItemRenderer();
```
Can you post a screenshot of what the whole thing looks like (or should look like)? Maybe we can do it a lot easier than that

eqsoft replied:
OK..
I will post some screenshots with different resolutions, because i was really impressed of the responsivness by the straight usage of the new UI components, so this LTI provider will work on a smartphone too, , thx!!

chfsx commented:
@eqsoft thanks for this PR! I would very much like to continue working on your branch myself, but unfortunately I don't know how to create an environment (locally) where I can see the changes. Do you have any simple instructions on how to do this in my existing development instance? Is there any kind of demo LTI content that I can easily include using configuration?

eqsoft committed:
Usability fixes for ltiprovider

eqsoft commented:
LTI Course view:
- MetaBar with Title of LTI object and exit button
- MainBar with single entry to LTI object
- Footer removed

![ltiprovider_crs1](https://user-images.githubusercontent.com/3586922/70855193-0508a800-1ec7-11ea-85d0-cbb5b0af823f.png)

LTI Course mobile:

![ltiprovider_crs2](https://user-images.githubusercontent.com/3586922/70855254-59605780-1ec8-11ea-9c08-195175a22853.png)

LTI Course mobile with unfolded exit slate  (nice):

![ltiprovider_crs3](https://user-images.githubusercontent.com/3586922/70855266-b0662c80-1ec8-11ea-8387-28e3b5c539af.png)

LTI Course as Course Admin and activated memberview:

![ltiprovider_crs4](https://user-images.githubusercontent.com/3586922/70855374-06879f80-1eca-11ea-9ecc-bc73d8e57779.png)

![ltiprovider_crs5](https://user-images.githubusercontent.com/3586922/70855377-1c956000-1eca-11ea-8060-adc59f43f79b.png)

nhaagen committed:
WIP: main-/metabar modifications
- add block for footer to main template
- change prios on MemberView/LTI-modificators
- make MemberViewModeInfo available
- add "withNoFooter" to standard-page

eqsoft commented:
this PR is based on a branch that is WIP so the discussion is not valid anymore -> close. 
I will push a new PR which is associated to the discussion of this PR.
